voltixApp/VoltixApp/View Models/Services/Thorchain/ThorchainService.swift

```diff
@@ -0,0 +1,186 @@
+    //
+    //  ThorchainService.swift
+    //  VoltixApp
+    //
+    //  Created by Enrique Souza Soares on 06/03/2024.
+    //
+
+import Foundation
+
+fileprivate struct BalanceCacheEntry: Codable {
+    let balances: [ThorchainBalance]
+    let timestamp: Date
+}
+
+fileprivate struct AccountCacheEntry: Codable {
+    let accountNumber: String
+    let timestamp: Date
+}
+
+class ThorchainService: ObservableObject {
+    static let shared = ThorchainService()
+    
+    @Published var balances: [ThorchainBalance]?
+    @Published var errorMessage: String?
+    @Published var accountNumber: String?
+    
+    func runeBalanceInUSD(usdPrice: Double?) -> String? {
+        guard let usdPrice = usdPrice,
+              let runeBalanceString = runeBalance,
+              let runeAmount = Double(runeBalanceString) else { return nil }
+        
+        
+        let balanceRune = runeAmount / 100_000_000.0
+        let balanceUSD = balanceRune * usdPrice
+        
+        let formatter = NumberFormatter()
+        formatter.numberStyle = .currency
+        formatter.locale = Locale.current
+        formatter.currencyCode = "USD"
+        
+        return formatter.string(from: NSNumber(value: balanceUSD))
+    }
+    
+    var formattedRuneBalance: String? {
+        guard let balances = balances else { return nil }
+        for balance in balances {
+            if balance.denom == "rune" {
+                guard let runeAmount = Double(balance.amount) else { return "Invalid balance" }
+                let balanceRune = runeAmount / 100_000_000.0
+                
+                let formatter = NumberFormatter()
+                formatter.numberStyle = .decimal
+                formatter.maximumFractionDigits = 8
+                formatter.minimumFractionDigits = 0
+                formatter.groupingSeparator = ""
+                formatter.decimalSeparator = "."
+                return formatter.string(from: NSNumber(value: balanceRune))
+            }
+        }
+        return "Balance not available"
+    }
+    
+    var runeBalance: String? {
+        guard let balances = balances else { return nil }
+        for balance in balances {
+            if balance.denom == "rune" {
```

Comment:
Is this checking for selected currency??
If so, can we use an enum instead? It gets prone to make a mistake here when using plain String.
We can remove any margin of error by using enums.

Reply:
It is not the selected coin, this is specific for the RUNE, but we can always create enums, not sure where should we place it. The coin at the moment is called THORCHAIN, but in the blockchain, it has another name. Maybe we can rename the coin to RUNE and keep the chain as Thorchain @johnnyluo?

Reply:
I will use the ticker instead @Rockindash. It will solve this issue